runner/init/init.go

```diff
@@ -59,19 +61,45 @@ func Exec(ctx context.Context) error {
 		logger.Info("cloning", "url", g.URL, "checkout", dfv1.PathCheckout)
 		var auth transport.AuthMethod
 
-		if k := g.SSHPrivateKey; k != nil {
-			logger.Info("getting secret for auth", "sshPrivateKey", k)
+		if k := g.UsernameSecret; k != nil {
+			if v := g.PasswordSecret; v != nil {
+				logger.Info("getting secret for auth", "UsernameSecret", k, "PasswordSecret", v)
+				secretInterface := kubernetes.NewForConfigOrDie(ctrl.GetConfigOrDie()).CoreV1().Secrets(os.Getenv(dfv1.EnvNamespace))
+
+				if usernameSecret, err := secretInterface.Get(ctx, k.Name, metav1.GetOptions{}); err != nil {
+					return fmt.Errorf("failed to get secret %q: %w", k.Name, err)
+				} else {
+					if pwdSecret, err := secretInterface.Get(ctx, v.Name, metav1.GetOptions{}); err != nil {
+						return fmt.Errorf("failed to get secret %q: %w", v.Name, err)
+					} else {
+						auth = &http.BasicAuth{
+							Username: string(usernameSecret.Data[v.Key]),
+							Password: string(pwdSecret.Data[k.Key]),
+						}
+					}
+				}
+			}
+		} else {
+			logger.Info("No username & password found for git auth.")
```

Comment:
minor - i think it should say "specified"  rather than "found"

Reply:
I just removed those log lines, since they don't bring value at the moment. I was just debugging with those :)